presto-docs/src/main/sphinx/connector/kafka.rst

```diff
@@ -259,6 +264,430 @@ Field           Required  Type      Description
+
+Example raw field definition in a :ref:`table definition file<kafka-table-definition-files>`
+for a Kafka message:
+
+.. code-block:: none
+
+    "message": {
+            "dataFormat": "raw",
+            "fields": [
+                {
+                    "name": "field1",
+                    "type": "BIGINT",
+                    "dataFormat": "LONG",
+                    "mapping": "0"
+                },
+                {
+                    "name": "field2",
+                    "type": "INTEGER",
+                    "dataFormat": "INT"
+                    "mapping": "8"
+                },
+                {
+                    "name": "field3",
+                    "type": "SMALLINT",
+                    "dataFormat": "LONG",
+                    "mapping": "12"
+                },
+                {
+                    "name": "field4",
+                    "type": "VARCHAR(6)",
+                    "dataFormat": "BYTE",
+                    "mapping": "20:26"
+                }
+            ]
+    }
+
+.. note::
+
+    Columns should be defined in the same order they are mapped. There can be
+    no gaps or overlaps between column mappings. The width of the column as
+    defined by the column mapping must be equivalent to the width of the
+    ``dataFormat`` for all types except for variable width types.
+
+Example ``INSERT INTO`` query for the above table definition:
+
+.. code-block:: SQL
+
+    INSERT INTO example_raw_table (field1, field2, field3, field4)
+      VALUES (123456789, 123456, 1234, 'abcdef');
+
+.. note::
+
+    When inserting variable width types the value must be exactly equal to the
+    width defined in the table definition file. If the inserted value is longer
+    it gets truncated, resulting in data loss. If the inserted value is shorter
+    the decoder will not be able to properly read the value because there is no
+    defined padding character. Due to these constraints the encoder fails if the
+    width of the inserted value is not equal to the mapping width.
+
+.. _csv-encoder:
+
+CSV encoder
+^^^^^^^^^^^^^^^
+
+The CSV encoder formats the values for each row as a line of
+comma-separated-values (CSV) using UTF-8 encoding.
+
+.. note::
+
+    The CSV line is formatted with a comma ',' as the column delimiter
+
+The ``type`` and ``mapping`` attributes must be defined for each field:
+
+* ``type`` - Presto data type
+* ``mapping`` - The integer index of the column in the CSV line (the first
+  column is 0, the second is 1, and so on)
+
+``dataFormat`` and ``formatHint`` are not supported and must be omitted.
+
+The following table lists supported Presto types, which can be used in ``type``
+and encoding scheme:
+
+============================= ================================================
+Presto data type              Encoding rules
+============================= ================================================
+``BIGINT``                    Formatted using Java ``Long.toString()``
```

Comment:
We shouldn't talk about Java implementation here. For integers, these are simply converted to strings. For floating point, these are converted to strings as if casted to varchar in Presto. Perhaps it's fine to simply say "converted to a string"